File: html/data/weapons.js
// Weapon data from wiki.aedelore.nu
// Note: Exact damage values not fully specified in wiki, using general guidelines

const WEAPONS_DATA = {
    // Simple Melee Weapons
    "Club": { type: "Simple Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "1", break: "1" },
    "Dagger": { type: "Simple Melee", ability: "Dexterity", bonus: "+1", damage: "1d6", range: "1", break: "0" },
    "Greatclub": { type: "Simple Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "2", break: "3" },
    "Handaxe": { type: "Simple Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "1", break: "2" },
    "Javelin": { type: "Simple Melee/Ranged", ability: "Strength", bonus: "+1", damage: "1d6", range: "3/10", break: "1" },
    "Light Hammer": { type: "Simple Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "1", break: "2" },
    "Mace": { type: "Simple Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "1", break: "2" },
    "Quarterstaff": { type: "Simple Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "3", break: "1" },
    "Sickle": { type: "Simple Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "1", break: "0" },
    "Spear": { type: "Simple Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "3", break: "1" },

    // Martial Melee Weapons
    "Battleaxe": { type: "Martial Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "1", break: "3" },
    "Flail": { type: "Martial Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "1", break: "2" },
    "Glaive": { type: "Martial Melee (Reach)", ability: "Strength", bonus: "+2", damage: "1d10", range: "3", break: "2" },
    "Greataxe": { type: "Martial Melee", ability: "Strength", bonus: "+3", damage: "2d6", range: "2", break: "5" },
    "Greatsword": { type: "Martial Melee", ability: "Strength", bonus: "+3", damage: "2d6", range: "2", break: "3" },
    "Halberd": { type: "Martial Melee (Reach)", ability: "Strength", bonus: "+2", damage: "1d10", range: "3", break: "3" },
    "Lance": { type: "Martial Melee (Reach)", ability: "Strength", bonus: "+2", damage: "2d6", range: "3", break: "2" },
    "Longsword": { type: "Martial Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "2", break: "2" },
    "Maul": { type: "Martial Melee", ability: "Strength", bonus: "+3", damage: "2d6", range: "2", break: "5" },
    "Morningstar": { type: "Martial Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "1", break: "3" },
    "Pike": { type: "Martial Melee (Reach)", ability: "Strength", bonus: "+2", damage: "1d10", range: "3", break: "2" },
    "Rapier": { type: "Martial Melee", ability: "Dexterity", bonus: "+2", damage: "1d10", range: "2", break: "0" },
    "Scimitar": { type: "Martial Melee", ability: "Dexterity", bonus: "+2", damage: "1d6", range: "1", break: "1" },
    "Shortsword": { type: "Martial Melee", ability: "Dexterity", bonus: "+2", damage: "1d6", range: "1", break: "1" },
    "Trident": { type: "Martial Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "3", break: "1" },
    "War Pick": { type: "Martial Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "1", break: "4" },
    "Warhammer": { type: "Martial Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "1", break: "4" },
    "Whip": { type: "Martial Melee (Reach)", ability: "Dexterity", bonus: "+1", damage: "1d6", range: "3", break: "0" },

    // Ranged Weapons
    "Crossbow (Light)": { type: "Ranged", ability: "Dexterity", bonus: "+2", damage: "1d6", range: "20/40", break: "0" },
    "Dart": { type: "Ranged", ability: "Dexterity", bonus: "+1", damage: "1d6", range: "5/15", break: "0" },
    "Shortbow": { type: "Ranged", ability: "Dexterity", bonus: "+2", damage: "1d6", range: "20/40", break: "0" },
    "Sling": { type: "Ranged", ability: "Dexterity", bonus: "+1", damage: "1d6", range: "10/30", break: "0" },
    "Blowgun": { type: "Ranged", ability: "Dexterity", bonus: "+1", damage: "1", range: "8/24", break: "0" },
    "Crossbow (Hand)": { type: "Ranged", ability: "Dexterity", bonus: "+2", damage: "1d6", range: "10/30", break: "0" },
    "Crossbow (Heavy)": { type: "Ranged", ability: "Strength", bonus: "+2", damage: "1d10", range: "30/60", break: "1" },
    "Longbow": { type: "Ranged", ability: "Dexterity", bonus: "+2", damage: "1d10", range: "40/80", break: "0" },
    "Net": { type: "Ranged", ability: "Dexterity", bonus: "+0", damage: "-", range: "2/5", break: "0" },

    // Fantasy Weapons
    "Falchion": { type: "Fantasy Melee", ability: "Strength", bonus: "+2", damage: "1d10", range: "1", break: "2" },
    "Claymore": { type: "Fantasy Melee", ability: "Strength", bonus: "+3", damage: "2d6", range: "2", break: "4" },
    "Katana": { type: "Fantasy Melee", ability: "Strength/Dexterity", bonus: "+2", damage: "1d10", range: "2", break: "2" },
    "Guisarme": { type: "Fantasy Melee (Reach)", ability: "Strength", bonus: "+2", damage: "1d10", range: "3", break: "3" },
    "Scythe": { type: "Fantasy Melee", ability: "Strength", bonus: "+1", damage: "1d10", range: "2", break: "1" },
    "Throwing Axe": { type: "Fantasy Ranged", ability: "Dexterity", bonus: "+2", damage: "1d6", range: "5/15", break: "1" },
    "Spiked Shield": { type: "Fantasy Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "1", break: "1" },
    "Chakram": { type: "Fantasy Ranged", ability: "Dexterity", bonus: "+2", damage: "1d6", range: "10/30", break: "0" },
    "Flamberge": { type: "Fantasy Melee", ability: "Strength", bonus: "+2", damage: "2d6", range: "2", break: "3" },
    "Bone Club": { type: "Fantasy Melee", ability: "Strength", bonus: "+1", damage: "1d6", range: "1", break: "1" }
};

// Simple array for datalist (backward compatibility)
const WEAPONS = Object.keys(WEAPONS_DATA);

// Ammunition pricing (from wiki)
const AMMUNITION = {
    "Arrows (20-50)": { cost: "1 gold" },
    "Bolts (20-50)": { cost: "1 gold" },
    "Blowgun Needles (20-50)": { cost: "1 gold" },
    "Sling Bullets (20)": { cost: "4 copper" }
};

File: android/app/src/main/java/nu/aedelore/app/domain/gamedata/Weapons.kt
package nu.aedelore.app.domain.gamedata

object Weapons {

    data class Weapon(
        val name: String,
        val type: String,
        val ability: String,
        val bonus: Int,
        val damage: String,
        val range: String,
        val breakValue: Int
    )

    val all: List<Weapon> = listOf(
        // Simple Melee
        Weapon("Club", "Simple Melee", "Strength", 1, "1d6", "1", 1),
        Weapon("Dagger", "Simple Melee", "Dexterity", 1, "1d6", "1", 0),
        Weapon("Greatclub", "Simple Melee", "Strength", 2, "1d10", "2", 3),
        Weapon("Handaxe", "Simple Melee", "Strength", 1, "1d6", "1", 2),
        Weapon("Javelin", "Simple Melee/Ranged", "Strength", 1, "1d6", "3/10", 1),
        Weapon("Light Hammer", "Simple Melee", "Strength", 1, "1d6", "1", 2),
        Weapon("Mace", "Simple Melee", "Strength", 1, "1d6", "1", 2),
        Weapon("Quarterstaff", "Simple Melee", "Dexterity", 1, "1d6", "3", 1),
        Weapon("Sickle", "Simple Melee", "Strength", 1, "1d6", "1", 0),
        Weapon("Spear", "Simple Melee", "Strength", 1, "1d6", "3", 1),

        // Martial Melee
        Weapon("Battleaxe", "Martial Melee", "Strength", 2, "1d10", "1", 3),
        Weapon("Flail", "Martial Melee", "Strength", 2, "1d10", "1", 2),
        Weapon("Glaive", "Martial Melee (Reach)", "Strength", 2, "1d10", "3", 2),
        Weapon("Greataxe", "Martial Melee", "Strength", 3, "2d6", "2", 5),
        Weapon("Greatsword", "Martial Melee", "Strength", 3, "2d6", "2", 3),
        Weapon("Halberd", "Martial Melee (Reach)", "Strength", 2, "1d10", "3", 3),
        Weapon("Lance", "Martial Melee (Reach)", "Strength", 2, "2d6", "3", 2),
        Weapon("Longsword", "Martial Melee", "Strength", 2, "1d10", "2", 2),
        Weapon("Maul", "Martial Melee", "Strength", 3, "2d6", "2", 5),
        Weapon("Morningstar", "Martial Melee", "Strength", 2, "1d10", "1", 3),
        Weapon("Pike", "Martial Melee (Reach)", "Strength", 2, "1d10", "3", 2),
        Weapon("Rapier", "Martial Melee", "Dexterity", 2, "1d10", "2", 0),
        Weapon("Scimitar", "Martial Melee", "Dexterity", 2, "1d6", "1", 1),
        Weapon("Shortsword", "Martial Melee", "Dexterity", 2, "1d6", "1", 1),
        Weapon("Trident", "Martial Melee", "Strength", 1, "1d6", "3", 1),
        Weapon("War Pick", "Martial Melee", "Strength", 2, "1d10", "1", 4),
        Weapon("Warhammer", "Martial Melee", "Strength", 2, "1d10", "1", 4),
        Weapon("Whip", "Martial Melee (Reach)", "Dexterity", 1, "1d6", "3", 0),

        // Ranged
        Weapon("Crossbow (Light)", "Ranged", "Dexterity", 2, "1d6", "20/40", 0),
        Weapon("Dart", "Ranged", "Dexterity", 1, "1d6", "5/15", 0),
        Weapon("Shortbow", "Ranged", "Dexterity", 2, "1d6", "20/40", 0),
        Weapon("Sling", "Ranged", "Dexterity", 1, "1d6", "10/30", 0),
        Weapon("Blowgun", "Ranged", "Dexterity", 1, "1", "8/24", 0),
        Weapon("Crossbow (Hand)", "Ranged", "Dexterity", 2, "1d6", "10/30", 0),
        Weapon("Crossbow (Heavy)", "Ranged", "Strength", 2, "1d10", "30/60", 1),
        Weapon("Longbow", "Ranged", "Dexterity", 2, "1d10", "40/80", 0),
        Weapon("Net", "Ranged", "Dexterity", 0, "-", "2/5", 0),

        // Fantasy
        Weapon("Falchion", "Fantasy Melee", "Strength", 2, "1d10", "1", 2),
        Weapon("Claymore", "Fantasy Melee", "Strength", 3, "2d6", "2", 4),
        Weapon("Katana", "Fantasy Melee", "Strength/Dexterity", 2, "1d10", "2", 2),
        Weapon("Guisarme", "Fantasy Melee (Reach)", "Strength", 2, "1d10", "3", 3),
        Weapon("Scythe", "Fantasy Melee", "Strength", 1, "1d10", "2", 1),
        Weapon("Throwing Axe", "Fantasy Ranged", "Dexterity", 2, "1d6", "5/15", 1),
        Weapon("Spiked Shield", "Fantasy Melee", "Strength", 1, "1d6", "1", 1),
        Weapon("Chakram", "Fantasy Ranged", "Dexterity", 2, "1d6", "10/30", 0),
        Weapon("Flamberge", "Fantasy Melee", "Strength", 2, "2d6", "2", 3),
        Weapon("Bone Club", "Fantasy Melee", "Strength", 1, "1d6", "1", 1)
    )

    fun byName(name: String): Weapon? = all.find {
        it.name.equals(name, ignoreCase = true)
    }
}
